src/postgres-util/src/tunnel.rs

```diff
@@ -226,7 +227,24 @@ impl Config {
         })?;
 
         match &self.tunnel {
-            TunnelConfig::Direct => {
+            TunnelConfig::Direct { resolved_ips } => {
+                if let Some(ips) = resolved_ips {
+                    let host = match postgres_config.get_hosts() {
+                        [Host::Tcp(host)] => host,
+                        _ => bail_generic!(
+                            "only TCP connections to a single PostgreSQL server are supported"
+                        ),
+                    }
+                    .to_owned();
+                    // The number of 'host' and 'hostaddr' values must be the same.
```

Comment:
```suggestion
                    // Associate each resolved ip with the exact same, singular host, for tls verification.
                    // We are required to do this dance because `tokio-postgres` enforces that the number of 'host' and 'hostaddr' values must be the same.
                    
```